tools/shared.py

```diff
@@ -520,9 +509,22 @@ def get_emscripten_version(path):
   return open(path).read().strip().replace('"', '')
 
 
+# The emscripten-version.txt file contains "dev" by default, indicating this is
+# a development version (probably a git clone of the emscripten repo). The
+# emsdk will update the file here with the actual version number when it
+# installs emscripten.
 EMSCRIPTEN_VERSION = get_emscripten_version(path_from_root('emscripten-version.txt'))
-parts = [int(x) for x in EMSCRIPTEN_VERSION.split('.')]
-EMSCRIPTEN_VERSION_MAJOR, EMSCRIPTEN_VERSION_MINOR, EMSCRIPTEN_VERSION_TINY = parts
+if '.' in EMSCRIPTEN_VERSION:
+  # A numeric X.Y.Z version number.
+  parts = [int(x) for x in EMSCRIPTEN_VERSION.split('.')]
+  EMSCRIPTEN_VERSION_MAJOR, EMSCRIPTEN_VERSION_MINOR, EMSCRIPTEN_VERSION_TINY = parts
+else:
+  # "dev", a git hash, or some other version indicator. We can't report a valid
+  # numeric version number here, so report 0.0.0.
+  EMSCRIPTEN_VERSION_MAJOR = 0
+  EMSCRIPTEN_VERSION_MINOR = 0
+  EMSCRIPTEN_VERSION_TINY = 0
```

Comment:
This does seems a little strange.  It is worth at least keeping the major version in source control still.  Something like "1.38.git"?  Or "1.38.dev"?